demo/rmm-integration/rmm_bridge.cu

```diff
@@ -0,0 +1,35 @@
+#include <vector>
+#include <string>
+#include <cstdio>
+
+#include <rmm/mr/device/cuda_memory_resource.hpp>
+#include <rmm/mr/device/pool_memory_resource.hpp>
+#include <rmm/mr/device/thread_safe_resource_adaptor.hpp>
+
+#include "rmm_bridge.h"
+
+using RMMCUDAMemoryResource = rmm::mr::cuda_memory_resource;
+using RMMPoolMemoryResource = rmm::mr::pool_memory_resource<RMMCUDAMemoryResource>;
+using RMMThreadSafePoolMemoryResource = rmm::mr::thread_safe_resource_adaptor<RMMPoolMemoryResource>;
```

Comment:
`pool_memory_resource` was recently made threadsafe on its own, so there's no need to use the thread safe adaptor.

Reply:
That will require users to use nightly version of RMM. Is it a common expectation?

Reply:
Perhaps add a TODO comment to remove it in whatever xgboost release corresponds to RMM 0.15.